pkg/certwatcher/certwatcher.go

```diff
@@ -208,7 +208,7 @@ func (cw *CertWatcher) ReadCertificate() error {
 		return nil
 	}
```

Comment:
Let's create a logger for CertWatcher in ~ l.68 and then use that one everywhere instead
```go
	cw := &CertWatcher{
		certPath: certPath,
		keyPath:  keyPath,
		interval: defaultWatchInterval,
		log:      log.WithValues("cert", certPath, "key", keyPath),
	}
```